Prog/Loans 2008-2017.sas

```diff
@@ -0,0 +1,418 @@
+	co_applicant_race_name_5 $
+	co_applicant_sex 
+	co_applicant_sex_name $
+	county_code 
+	county_name $
+	denial_reason_1 
+	denial_reason_2 
+	denial_reason_3 
+	denial_reason_name_1 $
+	denial_reason_name_2 $
+	denial_reason_name_3 $
+	edit_status 
+	edit_status_name $
+	hoepa_status 
+	hoepa_status_name $
+	lien_status 
+	lien_status_name $
+	loan_purpose 
+	loan_purpose_name $
+	loan_type 
+	loan_type_name $
+	msamd 
+	msamd_name $
+	owner_occupancy 
+	owner_occupancy_name $
+	preapproval 
+	preapproval_name $
+	property_type 
+	property_type_name $
+	purchaser_type 
+	purchaser_type_name $
+	respondent_id $
+	sequence_number 
+	state_code 
+	state_abbr $
+	state_name $
+	hud_median_family_income 
+	loan_amount_000s 
+	number_of_1_to_4_family_units 
+	number_of_owner_occupied_units 
+	minority_population 
+	population 
+	rate_spread 
+	tract_to_msamd_income 
+	;
+
+	if as_of_year=&year. ;
+	if census_tract_number ^= " ";
+
+run;
+
+
+/* Process a clean HMDA file */
+data hmda_&state._&year._clean;
+	set hmda_&state._&year._raw;
+
+	/* Set length for variables */
+	length action $1.
+	 	   agency $1.
+		   appdate $1.
+		   appdate $1.
+		   apprac $1.
+		   appsex $1.
+		   coapethn $1.
+		   coaprac $1.
+		   coaprac2 $1.
+		   coaprac3 $1.
+		   coaprac4 $1.
+		   coaprac5 $1.
+		   coapsex $1.
+		   deny1 $1.
+		   deny2 $1.
+		   deny3 $1.
+		   edit $1.
+		   ethn $1.
+		   hoepa $1.
+		   lien $1.
+		   metro $5.
+		   occupanc $1.
+		   prch_typ $1.
+		   preapp $1.
+		   purpose $1.
+		   race2 $1.
+		   race3 $1.
+		   race4 $1.
+		   race5 $1.
+		   type $1.
+		   county $3.
+		   ucounty $5.
+		   year $4.
+		   state $2.
+		   tract $6.
+		   high_interest 3
+	;
+
+	/* Create clean versions of variables */
+	action = put(action_taken,1.);
+	agency = put(agency_code,1.);
+	amount = loan_amount_000s * 1000;
+	appdate = put(application_date_indicator,1.);
+	apprac = put(applicant_race_1,1.);
+	appsex = put(applicant_sex,1.);
+	coapethn = put(co_applicant_ethnicity,1.);
+	coaprac = put(co_applicant_race_1,1.);
+	coaprac2 = put(co_applicant_race_2,1.);
+	coaprac3 = put(co_applicant_race_3,1.);
+	coaprac4 = put(co_applicant_race_4,1.);
+	coaprac5 = put(co_applicant_race_5,1.);
+	coapsex = put(co_applicant_sex,1.);
+	deny1 = put(denial_reason_1,1.);
+	deny2 = put(denial_reason_1,1.);
+	deny3 = put(denial_reason_1,1.);
+	edit = put(edit_status,1.);
+	ethn = put(applicant_ethnicity,1.);
+	hoepa = put(hoepa_status,1.);
+	hudmdinc = hud_median_family_income;
+	income = applicant_income_000s * 1000;
+	lien = put(lien_status,1.);
+	loantype = put(loan_type,1.);
+	metro = put(msamd,5.);
+	occupanc = put(owner_occupancy,1.);
+	prch_typ = put(purchaser_type,1.);
+	preapp = put(preapproval,1.);
+	purpose = put(loan_purpose,1.);
+	race2 = put(applicant_race_2,1.);
+	race3 = put(applicant_race_3,1.);
+	race4 = put(applicant_race_4,1.);
+	race5 = put(applicant_race_5,1.);
+	rtspread = rate_spread;
+	seq = sequence_number;
```

Comment:
The value of _seq_ is entirely missing in 2017.

Reply:
I've done some digging but am not 100% clear on why the seq variable is missing sometimes (and all the time in 2017). Regardless, it seems like this variable is not used in the summary program, so if it's OK with you I'd like to finalize this and move on to updating the summary files.